Modal.ts: `component` type change in `open`,`openInside`

CaselIT opened the pull request:
Just a nitpick.
In `open` and `openInside` changed the type of the argument `component` from `FunctionConstructor` to `Type` to avoid the need to cast an angular component to any.

Eg: [Here](https://github.com/shlomiassaf/angular2-modal/blob/master/src/demo/app/demoPage/demoPage.ts#L68) there would be no need to cast the `AdditionCalculateWindow` to `any`.

Also why not set `bindings` optional? There is already a check in the [code](https://github.com/shlomiassaf/angular2-modal/blob/master/src/components/angular2-modal/providers/Modal.ts#L95).
If you agree I can push a new change to this pr.

shlomiassaf commented:
You can push a commit with optional `bindings`, remember to apply a check since `config` is also optional.

CaselIT commented:
I've changed `bindings` to optional. I've used function overloads to allow both cases, and added the check in `openInside` to control which of the signatures was used.
I'm not sure if for the check is better to use `if(!Array.isArray(bindings))` (as is now) or `if(bindings instanceof ModalConfig)`

shlomiassaf commented:
Hi, Released version 1.0
It was too much to resolve the conflicts but I implemented your proposal and bindings are not required now.

Thanks!
